lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -63,7 +63,7 @@ func (d *zfs) CreateVolume(vol Volume, filler *VolumeFiller, op *operations.Oper
 
 			// For block volumes check if the cached image volume is larger than the current pool volume.size
 			// setting (if so we won't be able to resize the snapshot to that the smaller size later).
-			if vol.contentType == ContentTypeBlock {
+			if vol.contentType == ContentTypeBlock || shared.IsTrue(vol.ExpandedConfig("zfs.block_mode")) {
```

Comment:
Lets have a private function on this driver called something like `isBlockBacking() bool` that contains this logic and then it can be used everywhere that we need to make decisions based on the config. Similar to https://github.com/lxc/lxd/blob/fd94c234331b265d6b2ea690b3c0db573a8349b8/lxd/storage/drivers/driver_lvm_utils.go#L38